lib/BackgroundJobs/StorageCrawlJob.php

```diff
@@ -110,9 +147,11 @@ protected function run($argument): void {
 				->andWhere($qb->expr()->eq('storage', $qb->createNamedParameter($storageId)))
 				->andWhere($qb->expr()->in('mimetype', $qb->createNamedParameter($mimeTypes, IQueryBuilder::PARAM_INT_ARRAY)))
 				->andWhere($qb->expr()->gt('filecache.fileid', $qb->createNamedParameter($lastFileId)))
+				->andWhere($qb->expr()->notIn('parent', $qb->createNamedParameter($ignoreFileids, IQueryBuilder::PARAM_INT_ARRAY)))
```

Comment:
If ignoreFileids exceeds 1000, an error occurs "More than 1000 expressions in a list are not allowed on Oracle".
MariaDB encounters an error but completes the process successfully.
We will change the implementation to foreach and orWhere instead of notIn.

Reply:
You could group $ignoreFileids into sub arrays and do a andWhere-notIn for each sub array.